lib/sycamore/sycamore/evaluation/pipeline.py

```diff
@@ -36,46 +41,60 @@ def __init__(
         else:
             self._query_executor = query_executor
         self._os_config = os_config
+        self._subtask_docs = subtask_docs
+        self._embedder = embedder
 
     def _add_filter(self, query_body: dict, filters: dict[str, str]):
-        hybrid_query_match = query_body["query"]["hybrid"]["queries"][0]
-        hybrid_query_match = {
-            "bool": {
-                "must": [hybrid_query_match],
-                "filter": [{"match_phrase": {f"{k}.keyword": filters[k]}} for k in filters],
-            }
-        }
-        query_body["query"]["hybrid"]["queries"][0] = hybrid_query_match
-        hybrid_query_neural = query_body["query"]["hybrid"]["queries"][1]
-        hybrid_query_neural["neural"]["embedding"]["filter"] = {
-            "bool": {"must": [{"match_phrase": {k: filters[k]}} for k in filters]}
-        }
-        query_body["query"]["hybrid"]["queries"][1] = hybrid_query_neural
+        hybrid_query_match = query_body["query"]["knn"]["embedding"]["filter"]["bool"]["must"]
+        for key, val in filters.items():
+            hybrid_query_match.append({"match_phrase": {key: val}})
+        query_body["query"]["knn"]["embedding"]["filter"]["bool"]["must"] = hybrid_query_match
```

Comment:
wait what happens if it's a hybrid query?

Reply:
replaced knn query with hybrid

this is the workflow:

DocSet of questions --> DocSet of all formulas to be solved --> DocSet of formulas: elements of each document are subtasks --> DocSet of formulas: elements of each document are subtask answers

results of subtask execution passed to the EvaluationPipeline
in EvaluationPipeline's `_build_opensearch_query`, relevant results of subtask execution are added to original question